src/figuresClasses.ts

```diff
@@ -1,11 +1,85 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Invalid value! Each side must be bigger than 0!');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error(
+        'The sum of any two sides must be greater than the third side!',
+      );
+    }
+  }
+
+  getArea(): number {
+    const halfPerimetr = (this.a + this.b + this.c) / 2;
+    const triangleArea = Math.sqrt(
+      halfPerimetr *
+      (halfPerimetr - this.a) *
+      ((halfPerimetr - this.b) * (halfPerimetr - this.c)),
+    );
+
+    return Math.floor(triangleArea * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: Shape = 'circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Invalid value! Radius must be bigger than 0!');
+    }
+  }
+
+  getArea(): number {
+    const circleArea = Math.PI * this.radius ** 2;
+
+    return Math.floor(circleArea * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape: Shape = 'rectangle';
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
```

Comment:
[CODE KNOWLEDGE] - As with the other classes, the Rectangle class constructor does not need to manually assign 'color', 'width', and 'height' to class properties. The 'public' keyword in the parameters takes care of this.